code/model/SiteTree.php

```diff
@@ -2643,19 +2609,18 @@ public function setMenuTitle($value) {
 	}
 	
 	/**
-	 * A flag provides the user with additional data about the current page status,
-	 * for example a "removed from draft" status. Each page can have more than one status flag.
-	 * Returns a map of a unique key to a (localized) title for the flag.
-	 * The unique key can be reused as a CSS class.
-	 * Use the 'updateStatusFlags' extension point to customize the flags.
+	 * A flag provides the user with additional data about the current page status, for example a "removed from draft"
+	 * status. Each page can have more than one status flag. Returns a map of a unique key to a (localized) title for
+	 * the flag. The unique key can be reused as a CSS class. Use the 'updateStatusFlags' extension point to customize
+	 * the flags.
 	 * 
 	 * Example (simple): 
-	 * "deletedonlive" => "Deleted"
+	 *   "deletedonlive" => "Deleted"
 	 * 
 	 * Example (with optional title attribute): 
-	 * "deletedonlive" => array('text' => "Deleted", 'title' => 'This page has been deleted')
+	 *   "deletedonlive" => array('text' => "Deleted", 'title' => 'This page has been deleted')
 	 *
-	 * @param Boolean $cached
+	 * @param bool $cached Whether to serve the fields from cache; false regenerate them
```

Comment:
I'm of the opinion that documentation in loosely typed languages should represent semantics and not how the function itself actually works.

Can you pass `null` as a parameter? Usually always. Should you document `null` as a parameter type? Only if `null` is one of the intended usages of that method.

For instance:

```
public function doSomething($input = null) {
    if ($input = null) {
        // Set $input to something calculated that can't be represented as a default value
```

In this case I would not document `null` as being an option because you would never need to pass `null` to the function. Instead I would document `$input` as an optional override and state the expected type.

However, if there was a second parameter I *would* document `null` because you should use that to access the second parameter while keeping the default value for the first.

However if the default value was, for instance, `false`, and there was a second parameter, I would not document `null` as an option even if `null` would be coerced to `false` within the function.

Basically, my rule of thumb is that documentation should be for the user, and not the documentation library or the IDE. The latter can both read the method signature and identify the technically legal types. The documentation should always reflect the *intended* uses rather than the *possible* uses.